### DISCUSSION TITLE
module 'oracledb' has no attribute 'init_oracle_client' - but why?

### DISCUSSION THREAD

**Discussion by drnicolas** [oracle/python-oracledb#319] [Q&A] [answered]
I am using oracledb on another Linux-Installation since long  and it work including init_....

Now, on an additional installation oracledb seems not to know this command.
To make everything even more complicated - it is a docker installation

I remeber, I had to fight this problem also on the other computer but  I cannot remeber what the solution was.

I installed oracledb with apt install python3-oracledb and also downloaded the instantclient and unpacked in /Usr/lib ...

changing the lib_dir path does not help nor change the error message

Can someone give me a hint?

EDIT: I just checked both installations: the working seems to have oracledb 2.0.0.b1 while the other has 1.2.1
How comes ?

**Comment by cjbj** [comment]
Don't call your test file 'oracledb.py'? !

```
$ python3 oracledb.py
Traceback (most recent call last):
  File "/Users/cjones/p/oracledb.py", line 15, in <module>
    import oracledb
  File "/Users/cjones/p/oracledb.py", line 17, in <module>
    oracledb.init_oracle_client()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: partially initialized module 'oracledb' has no attribute 'init_oracle_client' (most likely due to a circular import)
```

**Reply by drnicolas** [reply]
okay, problem is partialyy solved:
Seems, the version available via apt install is too old.

"python3 -m pip install oracledb --upgrade" did not work for me, because there was a complaint that this package is externally managaed.
Finally, I found a workaround and then a more recent veriosn was installed - which has init_*

so far, so good.
but now it complains that "libnzz.so" was not found ( I passed lib_dir='/usr/lib/oracle/19.6/client64/lib'). If I change the path, then the error message is different.
Error message: " DPI-1047: Cannot locate a 64-bit Oracle Client library: "libnnz19.so: cannot open shared object file: No such file or directory". See https://python-oracledb.readthedocs.io/en/latest/user_guide/initialization.html for help"

I have no idea, as in this directory I see the file mentioned.

**Reply by cjbj** [reply]
You must set the library search path before running Python.  E.g. via LD_LIBRARY_PATH or running ldconfig.

**Reply by cjbj** [reply, marked as answer]
See https://python-oracledb.readthedocs.io/en/latest/user_guide/initialization.html#enabling-python-oracledb-thick-mode-on-linux-and-related-platforms
